apps/rig_inbound_gateway/lib/rig_inbound_gateway/api_proxy/router.ex

```diff
@@ -171,24 +193,26 @@ defmodule RigInboundGateway.ApiProxy.Router do
   end
 
   # Send error message with unsupported HTTP method
-  @spec send_response({:ok, %Plug.Conn{}, nil}) :: %Plug.Conn{}
+  @spec send_response({:ok, Plug.Conn.t, nil}) :: Plug.Conn.t
   defp send_response({:ok, conn, nil}) do
     send_resp(conn, 405, Serializer.encode_error_message("Method is not supported"))
   end
 
   # Send fulfilled response back to client
-  @spec send_response({:ok, %Plug.Conn{}, map}) :: %Plug.Conn{}
+  @spec send_response({:ok, Plug.Conn.t, map}) :: Plug.Conn.t
   defp send_response({:ok, conn, %{headers: headers, status_code: status_code, body: body}}) do
-    conn = %{conn | resp_headers: headers}
-    if Serializer.header_value?(conn, "transfer-encoding", "chunked") do
-      send_chunked_response(conn, headers, status_code, body)
+    downcased_headers = headers |> Serializer.downcase_headers
+    conn = %{conn | resp_headers: downcased_headers}
+
+    if Serializer.header_value?(downcased_headers, "transfer-encoding", "chunked") do
+      send_chunked_response(conn, downcased_headers, status_code, body)
```

Comment:
why are the headers passed here again - `conn` should already include them at this point

Reply:
*facepalm*